File: 1657-determine-if-two-strings-are-close/1657-determine-if-two-strings-are-close.py
class Solution:
  def closeStrings(self, word1: str, word2: str) -> bool:
    l1 = len(word1)
    l2 = len(word2)
    
    if l1 != l2:
      return False
    
    c1 = Counter(word1)
    c2 = Counter(word2)
    
    v1 = sorted(list(c1.values()))
    v2 = sorted(list(c2.values()))
    
    return set(word1) == set(word2) and v1 == v2

File: 1657-determine-if-two-strings-are-close/1657-determine-if-two-strings-are-close.cpp
class Solution {
public:
  bool closeStrings(string word1, string word2) {
    vector<int> f1(26), f2(26);
    for (char c : word1) ++f1[c-'a'];
    for (char c : word2) ++f2[c-'a'];
    return multiset(f1.begin(), f1.end()) == multiset(f2.begin(), f2.end()) && unordered_set(word1.begin(), word1.end()) == unordered_set(word2.begin(), word2.end());
  }
};